runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkPipelineOptions.java

```diff
@@ -146,11 +149,12 @@
    */
   @Description(
       "Sets the state backend to use in streaming mode. "
-          + "Otherwise the default is read from the Flink config.")
+          + "Otherwise the default is read from the Flink config. "
+          + "This should be an instance of {@link org.apache.flink.runtime.state.StateBackend}")
   @JsonIgnore
-  StateBackend getStateBackend();
+  Object getStateBackend();
 
-  void setStateBackend(StateBackend stateBackend);
+  void setStateBackend(Object stateBackend);
```

Comment:
This is kind of odd. We might break this into a factory method which takes the backend name and the parameters as a string. That way it would make it usable from the command-line.

Reply:
I tried to do this, but I didn't quite know how to let folks configure the backend nicely. The backends in flink have different constructor arguments which makes something generic...a little difficult. The factory gets the options object, so presumably it can be somewhat dynamic. Let me know if you were thinking of something different.